prototype/FFT_image_processing/arduino_image_processing/arduino_serial_port.cpp

```diff
@@ -0,0 +1,64 @@
+#include "arduino_serial_port.h"
+
+
+/**
+ * @brief Initialize serial
+ * @param baudRate The desired baudrate
+ * @param config The serial configuration
+*/
+void setupSerial(const unsigned long baudRate) {
+  Serial.begin(baudRate, SERIAL_8N1);
+}
+
+/**
+ * @brief Read an array of length arraySize, where each element is of size
+ * sizeOfElement. This function will block untill all elements are read.
+ * @param arraySize The size of the array to read.
+ * @param outArray The array in which to store the result of the read data.
+ * @param sizeOfElement The size of each element in the array.
+ */
+int readArray(const uint32_t arraySize, void* outArray,
+              const uint32_t sizeOfElement) {
+  readElement(outArray, arraySize * sizeOfElement);
+}
+
+/**
+ * @brief Write an array of length arraySize, where each element is of size
+ * sizeOfElement. This function will block untill all elements are written.
+ * @param arraySize The size of the array to read.
+ * @param array The array containing the data to be written.
+ * @param sizeOfElement The size of each element in the array.
+ */
+void writeFloatArray(const uint32_t arraySize, float* array,
```

Comment:
I don't see why we need this method. Can't you directly call writeArray with a pointer to your array and a size of sizeOfElement * arraySize ?

Reply:
Also, do we really need the parameter sizeOfElement if we know that this is an array of float ?